internal/github/publish_github_gist.go

```diff
@@ -0,0 +1,202 @@
+package github
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"log/slog"
+
+	gh "github.com/google/go-github/v58/github"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/zclconf/go-cty/cty"
+	"go.opentelemetry.io/otel/attribute"
+	nooptrace "go.opentelemetry.io/otel/trace/noop"
+
+	"github.com/blackstork-io/fabric/pkg/diagnostics"
+	"github.com/blackstork-io/fabric/plugin"
+	"github.com/blackstork-io/fabric/plugin/dataspec"
+	"github.com/blackstork-io/fabric/plugin/dataspec/constraint"
+	"github.com/blackstork-io/fabric/print"
+	"github.com/blackstork-io/fabric/print/htmlprint"
+	"github.com/blackstork-io/fabric/print/mdprint"
+)
+
+func makeGithubGistPublisher(loader ClientLoaderFn) *plugin.Publisher {
+	return &plugin.Publisher{
+		Doc:  "Publishes content to github gist",
+		Tags: []string{},
+		Config: dataspec.ObjectSpec{
+			&dataspec.AttrSpec{
+				Name:        "github_token",
+				Type:        cty.String,
+				Constraints: constraint.RequiredNonNull,
+				Secret:      true,
+			},
+		},
+		Args: dataspec.ObjectSpec{
+			&dataspec.AttrSpec{
+				Name: "description",
+				Type: cty.String,
+			},
+			&dataspec.AttrSpec{
+				Name: "filename",
+				Type: cty.String,
+			},
+			&dataspec.AttrSpec{
+				Name: "make_public",
+				Type: cty.Bool,
+			},
+			&dataspec.AttrSpec{
+				Name: "gist_id",
+				Type: cty.String,
+			},
+		},
+		AllowedFormats: []plugin.OutputFormat{plugin.OutputFormatMD, plugin.OutputFormatHTML},
+		PublishFunc:    publishGithubGist(loader),
+	}
+}
+
+func parseContent(data plugin.MapData) (document *plugin.ContentSection) {
+	documentMap, ok := data["document"]
+	if !ok {
+		return
+	}
+	contentMap, ok := documentMap.(plugin.MapData)["content"]
+	if !ok {
+		return
+	}
+	content, err := plugin.ParseContentData(contentMap.(plugin.MapData))
+	if err != nil {
+		return
+	}
+	document = content.(*plugin.ContentSection)
+	return
+}
+
+func parseName(data plugin.MapData) (name string) {
+	documentMap, ok := data["document"]
+	if !ok {
+		return
+	}
+	metaMap, ok := documentMap.(plugin.MapData)["meta"]
+	if !ok {
+		return
+	}
+	documentName := metaMap.(plugin.MapData)["name"]
```

Comment:
This is the wrong name to use - the name in the meta block is the template's name, not the document name. We should use the name of the `document` block (I think it is available?) -- "example" in the snippet above

Reply:
@traut document name wasn't available in params. added it now: https://github.com/blackstork-io/fabric/pull/222/commits/22cba2a5cf4fd4937e437dee6be91178c1907081